generate.py

```diff
@@ -84,40 +94,88 @@ def generate_api_collection() -> None:
                 request_item = {
-                    "name": method["name"],
+                    "name": method_name,
                     "request": {
-                        "method": method["httpmethod"],
+                        "method": method_type,
                         "url": {
-                            "raw": "https://api.steampowered.com",
-                            "host": ["api", "steampowered", "com"]
-                        },
-                        "header": [],
-                        "body": {
-                            "mode": "formdata",
-                            "formdata": []
+                            "protocol": "https",
+                            "host": ["api", "steampowered", "com"],
+                            "path": [interface_name, method_name, f'v{method["version"]}', ''],
+                            "query": []
                         },
-                        "description": method.get("description", "")
+                        "header": [
+                            {
+                                "key": "Content-Type",
+                                "value": "application/x-www-form-urlencoded"
+                            },
+                            # https://partner.steamgames.com/doc/webapi_overview/auth
+                            # just presume it's needed
+                            {
+                                "key": "x-webapi-key",
+                                "value": "{{key}}"
+                            }
+                        ],
+                        "description": method.get("description", ""),
                     }
                 }
 
+                items = []
+
                 # Add parameters as request parameters
                 for param in method["parameters"]:
+                    if param["name"] == "key":
+                        continue
+
                     param_item = {
                         "key": param["name"],
                         "value": "",
-                        "description": param.get("description", "")
+                        "description": param.get("description", ""),
+                        "disabled": param.get("optional", False)
                     }
 
-                    if param["type"] == "uint64":
-                        param_item["type"] = "number"
-                    elif param["type"] == "string":
+                    if method_type == 'POST':
+                        # formdata property type is always text
                         param_item["type"] = "text"
 
-                    request_item["request"]["body"]["formdata"].append(param_item)
+                    items.append(param_item)
+
+                if is_service_interface:
+                    input_dict = {i["key"]: i["value"] for i in items if not i.get("disabled", True)}
+                    request_item["request"]["description"] = f"{request_item["request"]["description"]}\n{
```

Comment:
Looks good, though I attempted to run this locally to verify it and Python is not happy with this f-string.

Reply:
Hrm, I wonder if this is a python version issue 
I was running on python 3.12. is there a lower bound you'd like me to test against?

I'll check it tonight.

Reply:
I fixed the f-string by utilizing triple quotes. So this works for python 3.11 and 3.12. Seems they updated the usage a little in 3.12: https://docs.python.org/3/whatsnew/3.12.html#pep-701-syntactic-formalization-of-f-strings